java/code/src/com/redhat/rhn/manager/formula/FormulaManager.java

```diff
@@ -127,6 +127,58 @@ public Map<String, Object> getGroupFormulaData(User user, String formulaName, Lo
         return data.orElse(Collections.emptyMap());
     }
 
+    /**
+     * Automatically convert String to Double and vice-versa to fulfill form type requirements
+     * @param content Content to be processed
+     * @param layout Form layout to be used for the content processing
+     */
+    public static void convertTypes(Map<String, Object> content, Map<String, Object> layout) {
```

Comment:
I think it might be useful to have a low-level log message (info/debug?) for converted values to catch any potential undesired conversions.